src/state.js

```diff
@@ -0,0 +1,27 @@
+import _ from 'lodash/fp'
+import * as F from 'futil-js'
+
+export let defaultState = {
+  path: null,
+  updating: null,
+  lastUpdateTime: null,
+  markedForUpdate: null,
+  hasValue: null,
+  error: null,
+  context: null,
+  missedUpdate: null,
+  pause: null,
+}
+
+export let setState = (flat, extend) =>
+  _.each(
+    node =>
+      extend(
+        node,
+        F.mapValuesIndexed(
```

Comment:
use `defaults` or something like that

Reply:
It should be like `_.pickBy(key => _.isNil(node[key]), defaultState)` but pickBy receives the value, not the key. Any thoughts?